### PR TITLE
Combine pkinit_login with get_ticket

### PR DESCRIPTION
**Pull request by zeroSteiner**
This updates the `auxiliary/admin/kerberos/get_ticket` module with PKINIT functionality, effectively combining it with `pkinit_login`. This will make it easier for users to go from a certificate directly to a TGS for service authentication.

## Verification

List the steps needed to make sure this thing works

- [ ] Start `msfconsole`
- [ ] Get a certificate as a user with the `icpr_cert` module
- [ ] `use kerberos/get_ticket`
- [ ] Test that you can issue a TGT with the certificate
- [ ] Test that you can issue a TGS with the certificate
- [ ] Try different option combinations, see that USERNAME and DOMAIN are required when the cert isn't present and are validated when it's not

### CODE REVIEW

**Comment by dwelch-r7** [comment]
Works for me

With cert:
image

without:
image

no spn:
image

missing password (etc.)
image

File: modules/auxiliary/admin/kerberos/get_ticket.rb [hunk 1]
```diff
@@ -73,6 +81,33 @@ def initialize(info = {})
   end
 
   def validate_options
+    if datastore['CERT_FILE'].present?
+      certificate = File.read(datastore['CERT_FILE'])
+      begin
+        @pfx = OpenSSL::PKCS12.new(certificate, datastore['CERT_PASSWORD'] || '')
```
Comment (dwelch-r7) [inline modules/auxiliary/admin/kerberos/get_ticket.rb:87]: Minor nitpick, ran into a small issue here when I set `CERT_PASSWORD` when one wasn't required:
image

pretty confusing error message, might be worth bubbling up the message from the exception here

Comment (zeroSteiner) [reply modules/auxiliary/admin/kerberos/get_ticket.rb:87]: That's a good idea. I added the error to the message now. Unfortunately though, when the password is incorrect, it doesn't state that but rather says there was a "mac verify failure". That's still probably more clear than saying it needs to be PKCS12 when it already is.

```
msf6 auxiliary(admin/kerberos/get_ticket) > rerun
[*] Reloading module...
[*] Running module against 192.168.159.10

[-] Auxiliary aborted due to failure: bad-config: Unable to parse certificate file (PKCS12_parse: mac verify failure)
[*] Auxiliary module execution completed
msf6 auxiliary(admin/kerberos/get_ticket) >
```

Comment (dwelch-r7) [reply modules/auxiliary/admin/kerberos/get_ticket.rb:87]: hah yea not ideal, but at least it's something

**Comment by dwelch-r7** [comment]
# Release Notes

Enhances the auxiliary/admin/kerberos/get_ticket module with PKINIT functionality

**Pull request merged**